app/components/SearchResults/AllResults/index.js

```diff
@@ -92,20 +93,20 @@ class AllResults extends React.Component {
   }
 
   render () {
-    if (
-      this.props.artistSearchResults.length <= 0 &&
-      this.props.albumSearchResults.length <= 0 &&
-      this.props.trackSearchResults.length <= 0
-    ) {
+    const tracksLength = _.get(this.props.trackSearchResults, ['info', 'length'], 0);
+    const artistsLength = _.get(this.props.artistSearchResults, ['length'], 0);
+    const albumsLength = _.get(this.props.albumSearchResults, ['length'], 0);
+    const playlistsLength = _.get(this.props.playlistSearchResults, ['info', 'length'], 0);
+    if (tracksLength + artistsLength + albumsLength + playlistsLength === 0) {
       return <div>{this.props.t('empty')}</div>;
     }
 
     return (
       <div className={styles.all_results_container}>
-        {this.renderArtistsSection()}
-        {this.renderAlbumsSection()}
-        {this.renderTracksSection()}
-        {this.renderPlaylistSection()}
+        {artistsLength > 0 && this.renderArtistsSection()}
```

Comment:
Maybe all these could be replaced with `_.isEmpty`? This way there would be no need to declare the consts above

Reply:
I think it won't be better, as `info` can be `null`, and checking `_.isEmpty(null)` is quite ambiguous - you have to remember the exact specification of Lodash call to tell if it's true or false. Moreover, `_.isEmpty` does some typechecking, which adds some performance overhead. So, I think the current solution is a little bit wordy, but readable, fast, and bulletproof.

Reply:
Ok, good point. However, you could simplify the syntax a bit like this:
```javascript
    const tracksLength = _.get(this.props.trackSearchResults, 'info.length', 0);
    const artistsLength = _.get(this.props.artistSearchResults, 'length', 0);
```

and similarly for the rest

Reply:
I tend to use arrays in Lodash get calls for three reasons:
1. It's faster as it does not involve any string parsing, and literal arrays can be extracted to constants to further improve the performance in critical paths.
2. It's magnitudes more reliable when you need dynamic expressions, especially when strings and numeric indices are mixed.
3. And taking in account  1 and 2, I use arrays in other cases for sake of consistency :)

Reply:
Nice, I didn't know that. I guess it can stay like that if there's a good reason for it.